R/end_rains.R

```diff
@@ -62,38 +60,38 @@ end_rains <- function(data, date_time, station = NULL, year = NULL, rain = NULL,
     grouping_by_station_year <- instatCalculations::instat_calculation$new(type="by", calculated_from=setNames(list(year), data))
-    doy_filter <- instatCalculations::instat_calculation$new(type="filter", function_exp="doy >= start_day & doy <= end_day", calculated_from=databook::calc_from_convert(x=list(data=doy)))
+    doy_filter <- instatCalculations::instat_calculation$new(type="filter", function_exp=paste0("doy >=", start_day, " & doy <= ", end_day), calculated_from=databook::calc_from_convert(x=setNames(list(doy), data)))
     
     # If "DOY (End Rains) is checked then run this line:
-    if (output == "day"){
+    if ("day" %in% output){
       end_rains <- instatCalculations::instat_calculation$new(type="summary", function_exp="ifelse(test=is.na(x=dplyr::last(x=roll_sum_rain)), yes=NA, no=dplyr::last(x=doy))", result_name="end_rains", calculated_from=setNames(list(doy), data), save=2)
     }
     # If Date (End Rains) is checked then run this line:
-    else if (output == "date"){
+    if ("date" %in% output){
       end_rains_date <- instatCalculations::instat_calculation$new(type="summary", function_exp="dplyr::if_else(condition=is.na(x=dplyr::last(x=roll_sum_rain)), true=as.Date(NA), false=dplyr::last(x=date_time))", result_name="end_rains_date", calculated_from=setNames(list(date_time), data), save=2)
     }
     # If Status (End Rains) is checked then run this line:
-    else {
+    if ("status" %in% output) {
       end_rains_status <- instatCalculations::instat_calculation$new(type="summary", function_exp="ifelse(n() > 0, yes=ifelse(is.na(x=dplyr::last(x=roll_sum_rain)), yes=NA, no=TRUE), no=FALSE)", result_name="end_rains_status", save=2)
     }
     
-    outputs_selected <- c("end_rains", "end_rains_date", "end_rains_status")
-    outputs_selected <- outputs_selected[sapply(outputs_selected, exists)]
-    outputs_selected <- mget(outputs_selected)
-    
+    # Now collect only the ones you created into a list:
+    sub_calcs <- list()
+    if (exists("end_rains"))         sub_calcs <- c(sub_calcs, list(end_rains))
+    if (exists("end_rains_date"))    sub_calcs <- c(sub_calcs, list(end_rains_date))
+    if (exists("end_rains_status"))  sub_calcs <- c(sub_calcs, list(end_rains_status))
+  
     # In Combined, we run all that were checked out of end_rains, end_rains_date, end_rains_status and put them in sub_calculations
-    end_of_rains_combined <- instatCalculations::instat_calculation$new(type="combination", manipulations=list(conditions_filter, grouping_by_station_year, doy_filter), sub_calculations=list(outputs_selected))
-    
+    end_of_rains_combined <- instatCalculations::instat_calculation$new(type="combination", manipulations=list(conditions_filter, grouping_by_station_year, doy_filter), sub_calculations = sub_calcs)
```

Comment:
I was having a few issues with your code, so I amended it (with the help of the robots) to this which is working for me.
I'm flagging here as this is probably also going to be how you want to do this in end_season

Reply:
You’re working not giving me work to do! You’re taking away work to do by doing it. Thank you very much, this is looking great and really nice. I hope this is making sense to you? It looks like you’re getting a good sense of the calculation system to me.

I’ll send you some data and a script tomorrow so you can test yourself - but otherwise this is no big deal. Only took 10 minutes for me to run through and make these changes.